### PR TITLE
Improve imagelist custom field

### PR DESCRIPTION
**Pull request by PhilETaylor**
Pull Request for Issue #16121 .

### Summary of Changes

Tidy up and bugfix custom field imagelist

### Testing Instructions

Create a new Custom Field List of Images for the Content. Create a new article and select an image for the field.
See the article in frontend and see the image path:
src="/Joomla_3.7.1/images**///**image-name.gif"

Apply patch

### Expected result

src="/Joomla_3.7.1/images/image-name.gif"

### NOTE
Coded without testing

### CODE REVIEW

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 252d94ec80213737281bd81d3d20d0e0454e39c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16135">issues.joomla.org/tracker/joomla-cms/16135</a>.</sub>

**Comment by PhilETaylor** [comment]
Please also take a moment to provide feedback on the output of this filed in this issue https://github.com/joomla/joomla-cms/issues/16136

File: plugins/fields/imagelist/tmpl/imagelist.php [hunk 1]
```diff
@@ -31,8 +30,17 @@
...
+	if ($fieldParams->get('directory', '/') !== '/')
+	{
+		$buffer .= sprintf('<img src="images/%s/%s" %s />',
+			$fieldParams->get('directory') ,
+			htmlentities($path, ENT_COMPAT, true),
+			$class);
+	}
+	else
+	{
+		$buffer .= '<img src="images/' . htmlentities($path) . '"' . $class . '/>';
```
Comment (Quy) [inline plugins/fields/imagelist/tmpl/imagelist.php]: Add a space before and after the `class` attribute.

Comment (PhilETaylor) [reply plugins/fields/imagelist/tmpl/imagelist.php]: No. This is not the `$class` attribute, this is a variable that contains something like ` class="this" ` and already contains the spaces need as per the above logic on line https://github.com/PhilETaylor/joomla-cms/blob/4a29649109a247070037d32cfa193726b554acba/plugins/fields/imagelist/tmpl/imagelist.php#L20

Comment (PhilETaylor) [reply plugins/fields/imagelist/tmpl/imagelist.php]: Actually Yes, Im wrong :) doing now ...

Comment (PhilETaylor) [reply plugins/fields/imagelist/tmpl/imagelist.php]: Corrected in cb5e207

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on cb5e207e59c7cbbd1c076fb1961f954de39eb44e<br /><br />Patch ok for me. Thanks.

Is it a good idea also add the ALT tag with the field Label text ?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16135">issues.joomla.org/tracker/joomla-cms/16135</a>.</sub>

**Comment by PhilETaylor** [comment]
> Is it a good idea also add the ALT tag with the field Label text ?
@AlexRed Please add that suggestion here: #16136

**Comment by Quy** [comment]
When there is no class, which is preferred? `"/>` or `" />"

**Comment by PhilETaylor** [comment]
I think WITH the space is "more right" ? 
Ref: https://www.tutorialspoint.com/xhtml/xhtml_syntax.htm

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 60b1ac1dedf8b2e77b2a31acfb00cbae2981700c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16135">issues.joomla.org/tracker/joomla-cms/16135</a>.</sub>

**Comment by brianteeman** [comment]
@c-lodder slashes?

**Comment by franz-wohlkoenig** [comment]
@AlexRed can you please retest?

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on 60b1ac1dedf8b2e77b2a31acfb00cbae2981700c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16135">issues.joomla.org/tracker/joomla-cms/16135</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

File: plugins/fields/imagelist/tmpl/imagelist.php [hunk 2]
```diff
@@ -31,8 +31,21 @@
 		continue;
 	}
 
-	$buffer .= '<img src="images/' . $fieldParams->get('directory', '/')
-				. '/' . htmlentities($path) . '"' . $class . '/>';
+	if ($fieldParams->get('directory', '/') !== '/')
+	{
+		$buffer .= sprintf('<img src="images/%s/%s"%s />',
```
Comment (C-Lodder) [inline plugins/fields/imagelist/tmpl/imagelist.php]: remove `/` at end of `` tag.

File: plugins/fields/imagelist/tmpl/imagelist.php [hunk 3]
```diff
@@ -31,8 +31,21 @@
...
+			$fieldParams->get('directory'),
+			htmlentities($path, ENT_COMPAT, 'UTF-8', true),
+			$class
+		);
+	}
+	else
+	{
+		$buffer .= sprintf('<img src="images/%s"%s />',
```
Comment (C-Lodder) [inline plugins/fields/imagelist/tmpl/imagelist.php]: same here

**Comment by PhilETaylor** [comment]
If you are going to enforce that then there a lot of other image outputs in Joomla that also need addressing!

**Comment by C-Lodder** [comment]
They've been all done in Joomla 4. I'm only asking on this so it doesn't need to be addressed when staging is merged into 4.0-dev

**Comment by dgt41** [comment]
@C-Lodder 3.x is *not* only HTML5...

**Comment by PhilETaylor** [comment]
So what you are really saying is that you accepted a new feature in Joomla, with substandard, insecure code that did not meet the [Joomla Coding Standards](https://developer.joomla.org/coding-standards/html.html)

and now you want to hold PR creators to a higher standard than you do those that contribute insecure buggy new features to Joomla.

Got it.

**Comment by PhilETaylor** [comment]
To be clear, this and media.php ALREADY HAD the closing slash before this PR was 'contributed'

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6aad8876170de1c1546890e38d655eb9075c137e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/16135">issues.joomla.org/tracker/joomla-cms/16135</a>.</sub>

**Pull request merged**